### PR TITLE
Implemented TemplateProvider.

### PR DESCRIPTION
**Pull request by Siddhanttimeline**
### Describe your changes:

Implemented TemplateProvider.

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
@Siddhanttimeline  @mohityadav766 can we not use the system settings to implement email templates and make them available through the settings in UI itself. Instead of using config in server side.

File: openmetadata-service/src/main/java/org/openmetadata/service/util/DefaultTemplateProvider.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  @Override
+  public Template getTemplate(String templateName) throws IOException {
+    return templateConfiguration.getTemplate(templateName);
+  }
+
+  public static String getTemplatePath() {
+    SmtpSettings emailConfig =
+        SettingsCache.getSetting(SettingsType.EMAIL_CONFIGURATION, SmtpSettings.class);
```
Comment (Siddhanttimeline) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/DefaultTemplateProvider.java:39]: @harshach We are reading the configurations from the SystemSettings only, which are being set from the UI. In the front end, we need to set this field templatePath and send its value along with the other configuration values.

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/DefaultTemplateProvider.java:39]: I don't want email settings to be a server side config at all. Lets drive everything through the APIs and add a ./openmetadata-ops.sh that can pickup environment variables and update the email configs.

Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/DefaultTemplateProvider.java:39]: Noted.

**Comment by Siddhanttimeline** [comment]
Adding the changes in different branch.

**Pull request closed without merging**